nightingale-server-prod.yaml

```diff
@@ -0,0 +1,198 @@
+        image: harbor.nightingale.revelry.org/library/nightingale-server:develop-c030c6e
+        ports:
+        - containerPort: 5000
+        envFrom:
+        - configMapRef:
+            name: nightingale-env-config
+        - secretRef:
+            name: nightingale-env-secret
+        - secretRef:
+            name: kube-pull
+        - secretRef:
+            # This secret is created by KubeDB along with the db instance
+            # The secret name follows the pattern {Postgres resource name}-auth
+            name: nightingale-database-auth
+        env:
+        # distributed elixir via libcluster
+        - name: CLUSTER_ENABLED
+          value: "1"
+        # let each instance know its own IP for distributed elixir
+        - name: NODE_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+      initContainers:
+      - name: migrate
+        image: harbor.nightingale.revelry.org/library/nightingale-server:master-ceb790f
+        command: ["./bin/nightingale"]
+        args: ["eval", "Nightingale.ReleaseTasks.migrate()"]
+        envFrom:
+        - configMapRef:
+            name: nightingale-env-config
+        - secretRef:
+            name: kube-pull
+        - secretRef:
+            # This secret is created by KubeDB along with the db instance
+            # The secret name follows the pattern {Postgres resource name}-auth
+            name: nightingale-database-auth
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nightingale
+  namespace: nightingale
+spec:
+  selector:
+    app: nightingale
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 5000
+---
+# This 'headless service' lets app nodes discover one another through DNS
+# enabling distributed elixir features
+apiVersion: v1
+kind: Service
+metadata:
+  name: nightingale-nodes
+  namespace: nightingale
+spec:
+  clusterIP: None
+  selector:
+    app: nightingale
+  ports:
+  - name: epmd
+    port: 4369
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: nightingale
+  namespace: nightingale
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    certmanager.k8s.io/cluster-issuer: letsencrypt-production
+spec:
+  rules:
+  - host: nightingale.revelry-prod.revelry.net
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: nightingale
+          servicePort: 80
+  tls:
+  - hosts:
+    - nightingale.revelry-prod.revelry.net
+    secretName: nightingale-cert
+---
+apiVersion: kubedb.com/v1alpha1
+kind: Postgres
+metadata:
+  name: nightingale-database
+  namespace: nightingale
+spec:
+  version: "11.2"
+  storageType: Durable
+  storage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 1Gi
+  terminationPolicy: DoNotTerminate
+  backupSchedule:
+    cronExpression: "@every 1h"
+    storageSecretName: nightingale-s3-secret
+    s3:
+      endpoint: "s3.amazonaws.com"
+      bucket: kubedb-revelry-prod
```

Comment:
@jwietelmann i think i'll need a bucket name for the backups

Reply:
@CurtisVonRubenhoff `rev-nightingale-kubedb`